File: common/src/main/java/com/common/DataClasses/Position.java
package com.common.DataClasses;

public class Position {
	public int x;
	public int y;

	public Position() {
		x = 0;
		y = 0;
	}

	public Position(int x, int y) {
		this.x = x;
		this.y = y;
	}

	public Position(Position position) {
		x = position.x;
		y = position.y;
	}

	public boolean equals(Position position) {
		return (position.x == x) && (position.y == y);
	}
}

File: composeApp/src/commonMain/kotlin/com/one_of_many_simons/one_button_game/dataClasses/Position.kt
package com.one_of_many_simons.one_button_game.dataClasses

class Position {
    var x: Int = 0
    var y: Int = 0

    constructor() {
        x = 0
        y = 0
    }

    constructor(x: Int, y: Int) {
        this.x = x
        this.y = y
    }

    constructor(position: Position) {
        this.x = position.x
        this.y = position.y
    }

    @Suppress("CovariantEquals")
    fun equals(position: Position): Boolean {
        return x == position.x && y == position.y
    }
}